modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-report-web/src/main/java/com/liferay/dynamic/data/mapping/form/report/web/internal/action/FormInstanceRecordsFieldValueMVCResourceCommand.java

```diff
@@ -0,0 +1,139 @@
+import com.liferay.dynamic.data.mapping.storage.DDMFormValues;
+import com.liferay.dynamic.data.mapping.util.comparator.DDMFormInstanceRecordModifiedDateComparator;
+import com.liferay.portal.kernel.dao.orm.QueryUtil;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.portlet.JSONPortletResponseUtil;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCResourceCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCResourceCommand;
+import com.liferay.portal.kernel.security.auth.PrincipalException;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ContentTypes;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.portlet.ResourceRequest;
+import javax.portlet.ResourceResponse;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Marcos André
+ */
+@Component(
+	immediate = true,
+	property = {
+		"javax.portlet.name=" + DDMPortletKeys.DYNAMIC_DATA_MAPPING_FORM_REPORT,
+		"mvc.command.name=/forms/records/field/value"
+	},
+	service = MVCResourceCommand.class
+)
+public class FormInstanceRecordsFieldValueMVCResourceCommand
+	extends BaseMVCResourceCommand {
+
+	@Override
+	protected void doServeResource(
+			ResourceRequest resourceRequest, ResourceResponse resourceResponse)
+		throws Exception {
+
+		HttpServletRequest httpServletRequest = _portal.getHttpServletRequest(
+			resourceRequest);
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		if (!themeDisplay.isSignedIn()) {
+			throw new PrincipalException.MustBeAuthenticated(
+				themeDisplay.getUserId());
+		}
+
+		JSONArray fieldValuesJSONArray = _getFieldValuesJSONArray(
+			httpServletRequest);
+
+		HttpServletResponse httpServletResponse =
+			_portal.getHttpServletResponse(resourceResponse);
+
+		httpServletResponse.setContentType(ContentTypes.APPLICATION_JSON);
+
+		JSONPortletResponseUtil.writeJSON(
+			resourceRequest, resourceResponse, fieldValuesJSONArray);
+	}
+
+	private JSONArray _getFieldValuesJSONArray(
+			HttpServletRequest httpServletRequest)
+		throws Exception {
+
+		JSONArray jsonArray = JSONFactoryUtil.createJSONArray();
+
+		String fieldName = ParamUtil.getString(httpServletRequest, "fieldName");
+
+		long formInstanceId = ParamUtil.getLong(
+			httpServletRequest, "formInstanceId");
+
+		int start = ParamUtil.getInteger(
+			httpServletRequest, "start", QueryUtil.ALL_POS);
+		int end = ParamUtil.getInteger(
+			httpServletRequest, "end", QueryUtil.ALL_POS);
```

Comment:
@marcosapmf entendo que essa task combinamos de retornar todos as linhas, sem paginação. se liga que quando adicionarmos paginação, o usuário não vai poder passar o `end` como parâmetro pois somos nós quem decidimos  o tamanho da página e não o client. o problema de deixar o usuário pegar todos as rows, é que pode deixar o servidor lento e dar denial of service caso existam milhares de records.